media-libs/assimp/assimp-5.4.3.ebuild

```diff
@@ -0,0 +1,127 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Importer library to import assets from 3D files"
+HOMEPAGE="https://github.com/assimp/assimp"
+SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+IUSE="collada doc samples test"
+
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	sys-libs/zlib[minizip]
+	collada? ( media-libs/opencollada )
+	doc? ( app-text/doxygen )
+	samples? (
+		media-libs/freeglut
+		media-libs/libglvnd
+	)
+	test? (
+		dev-cpp/gtest
+	)
+"
+DEPEND="${RDEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.2.5-gtest.patch"
+	"${FILESDIR}/${PN}-5.2.5-float-comparison.patch"
+)
+
+DOCS=( CodeConventions.md Readme.md )
+
+src_prepare() {
+	sed -r \
+		-e "s#PROJECT\(Assimp VERSION [0-9]+\.[0-9]+\.[0-9]+\)#PROJECT(Assimp VERSION ${PV})#g" \
+		-i CMakeLists.txt || die
+	sed -r \
+		-e "s#EXPECT_EQ\(aiGetVersionPatch\(\), [0-9]+U \);#EXPECT_EQ(aiGetVersionPatch(), $(ver_cut 3)U );#g" \
+		-e "s#EXPECT_NE\( aiGetVersionRevision#EXPECT_EQ\( aiGetVersionRevision#g" \
+		-i test/unit/utVersion.cpp || die
```

Comment:
```
 * SED: the following did not cause any changes
 *     sed -r -e "s#PROJECT\(Assimp VERSION [0-9]+\.[0-9]+\.[0-9]+\)#PROJECT(Assimp VERSION ${PV})#g" -i CMakeLists.txt || die;
 * no-op: -e s#PROJECT\(Assimp VERSION [0-9]+\.[0-9]+\.[0-9]+\)#PROJECT(Assimp VERSION 5.4.3)#g
 * SED: the following did not cause any changes
 *     sed -r -e "s#EXPECT_EQ\(aiGetVersionPatch\(\), [0-9]+U \);#EXPECT_EQ(aiGetVersionPatch(), $(ver_cut 3)U );#g" -e "s#EXPECT_NE\( aiGetVersionRevision#EXPECT_EQ\( aiGetVersionRevision#g" -i test/unit/utVersion.cpp || die;
 * no-op: -e s#EXPECT_EQ\(aiGetVersionPatch\(\), [0-9]+U \);#EXPECT_EQ(aiGetVersionPatch(), 3U );#g
 * no-op: -e s#EXPECT_NE\( aiGetVersionRevision#EXPECT_EQ\( aiGetVersionRevision#g
```
I always appreciate a comment above any sed explaining what it does, since it lets me make a decision whether I can remove the said sed while merging, or whether it's something that **needs** to work before merging.